File: projects/04/compute_factorial.rb
#function computeFactorial returns the factorial of a number passed
#parameters: n, the number to be factoralized
#precondition: n is a positive integer
#return: the Factoralization of n.
#2/24/18
#Completed by: Bryce Allen for cs 214 at Calvin College
##############################################################################

def computeFactorial(n)
  a = 1.0
  for i in 2..n
    a = a*i
  end
  return a
end


if __FILE__ == $0
   print " enter a number: "
   num = gets.chomp.to_f
   print "Factorialization: "
   puts computeFactorial(num)
end

File: projects/04/ComputeFactorial.java
/*ComputeFactorial.java
 *Computes factorial with user friendly implementation
 *By: Bryce Allen
 *cs 214
 *Calvin College
 *2/23/18
 */
import java.util.Scanner;

public class ComputeFactorial
{
    public static void main(String[] args)
    {
	Scanner keyboard = new Scanner(System.in);
	System.out.println("Enter a value: ");
	int number = keyboard.nextInt();
	double factorial = factorial(number);
	System.out.println("The factorial of " + number + " is " + factorial);
    }

    /*written by: Bryce Allen.
     *@param: n representing the integer to be factoralized.
     *@return: the factorialized answer.
     */
    public static double factorial(int n)
    {
	double answer = 1.0;
	for(int i = 2; i <= n; i++)
        {  answer *= i; }
	return answer;
    }
}
